source/MaterialXTest/GenShaderUtil.cpp

```diff
@@ -677,53 +677,59 @@ void ShaderGeneratorTester::validate(const mx::GenOptions& generateOptions, cons
                 nodeDef = shaderRef->getNodeDef();
             }
 
-            // Allow to skip nodedefs to test if specified
-            const std::string nodeDefName = nodeDef->getName();
-            if (_skipNodeDefs.count(nodeDefName))
+            if (!nodeDef)
             {
-                _logFile << ">> Skipped testing nodedef: " << nodeDefName << std::endl;
-                continue;
+                CHECK(nodeDef);
```

Comment:
Won't this just always fail and the skipping logic ignored?

Reply:
This will always fail if it gets into the if, yes. It's just to show there's an issue.